README.md

```diff
@@ -121,6 +121,27 @@ Without the `--abort` option, the entire KB will be validated with full output o
 To fail fast when using the validator programmatically, use `ValidateAIF.setAbortThreshold()` to set an error
 threshold.
 
+### Shallow validation
+
+The `--depth` option performs a "shallow" validation in which each SHACL rule (shape) only considers a subset of its
+target nodes.  The size of this subset (i.e., the depth of the validation) can be specified on the command line.
+For example, `--depth=100` means that if your file has 30,000 event arguments, then the `aida:EventArgumentShape` will
+only be applied to 100 event arguments, significantly speeding up generation of an error report.  Any violations in
+these 100 nodes will be included in the error report.  By default (if no depth is specified), only 50 target nodes will
+be tested.  The `--depth` option requires enabling the multi-threaded validator via the `-t` option.
+Unlike failing fast, shallow validation ends early whether or not it finds any SHACL violations.
+
+To enable shallow validation programmatically, use `ValidateAIF.setDepth()` and specify a depth.
+
+### Memory considerations
+
+Validation of a large files can require significant system resources, particularly system RAM.  By default, the Java
+heap will use up to half of the available RAM on your system.  If you want to set a higher (or lower) maximum, then
+set the `JAVA_OPTS` environment variable to, for example, `-mx16G` to use up to 16GB of RAM.
```

Comment:
Shouldn't `-mx16G` be `-Xmx16G` ?

Reply:
According to [StackOverflow](https://stackoverflow.com/questions/5374455/what-does-java-option-xmx-stand-for), options which start with -X are non standard across platforms and may be removed in the future. There is a -mx which is standard, shorter and does the same thing.

That said, I'd never heard of it, and have always used -Xmx.  It's probably better-known overall.  Would you rather I switch to -Xmx?  I don't feel strongly either way.

Reply:
Fixed.